apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java

```diff
@@ -84,11 +105,59 @@ public Span withName(@Nullable String name) {
         return this;
     }
 
+    /**
+     * Keywords of specific relevance in the span's domain (eg: 'db', 'template', 'ext', etc)
+     */
+    public Span withType(@Nullable String type) {
+        if (!isSampled()) {
```

Comment:
I think checking for sampled is not really necessary, I'd remove it. It can lead to problems (see #447). If we want to track metrics for spans we also might expect the type to be set.

Reply:
removing for `withSubtype` and `withAction` as well